skyportal/models/instrument.py

```diff
@@ -43,10 +55,21 @@ def process(value):
         return process
 
 
+def manage_instrument_access_logic(cls, user_or_token):
+    if user_or_token.is_system_admin:
+        return DBSession().query(cls)
+    elif 'Manage allocations' in [acl.id for acl in user_or_token.acls]:
+        return DBSession().query(cls)
+    else:
+        # return an empty query
+        return DBSession().query(cls).filter(cls.id == -1)
```

Comment:
Does this style of empty query appear elsewhere? Kind of interesting.

Reply:
That's the problem, I didn't find anywhere else where that was done. In all other handler it would join the table to another, then to another .... until it reaches somehow the acls to filter on that, or join the table on itself, ... I haven't found a clear example of how to return an empty query. I tried writing it that way as it made sense to me and seemed simpler to understand. But I can change that if it seems incorrect.

Reply:
@Theodlz honestly it makes sense to me, I just wasn't sure if there was already precedent of some kind!